File: Booth.DockerVolumeBackup.Domain/Backup.cs

namespace Booth.DockerVolumeBackup.Domain
{
    public enum Status { Queued, Active, Complete, Error }

    public class Backup
    {
        public int BackupId { get;set; }
        public Status Status { get;set; }
        public int? ScheduleId { get ;set; }
        public DateTimeOffset ScheduledTime { get;set; }
        public DateTimeOffset StartTime { get; set; }
        public DateTimeOffset EndTime { get; set; }
    }
}

File: booth.dockervolumebackup.ui/src/models/Backup.ts

export type BackupType = 'Scheduled' | 'Adhoc' | 'Unmanaged';
export type Status = 'Queued' | 'Active' | 'Complete' | 'Error';

export interface Backup {
    backupId: number;
    backupType: BackupType;
    scheduleId: number;
    scheduleName: string;
    status: Status;
    backupTime: Date;
}

export interface BackupDetail {
    backupId: number
    backupType: BackupType;
    scheduleId?: number;
    scheduleName: string;
    status: Status;
    startTime?: Date;
    endTime?: Date;
    volumes: BackupVolumeDetail[];
}

export interface BackupVolumeDetail {
    backupVolumeId: number;
    volume: string;
    status: Status;
    startTime?: Date;
    endTime?: Date;
    backupSize?: number;
}
